proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+
+While task groups are indeed very powerful, they are hard to use with *heterogenous results* and step-by-step initialization patterns. 
+
+The following example, an asynchronous `makeDinner` function, consists of both of those patterns. It consists of three tasks which can be performed in parallel, all yielding different result types. To proceed to the final step of the cooking process, all those results need to be obtained, and fed into the final `oven.cook(...)` function. In a way, this is the trickiest situation to implement well using task groups. Let us examine it more closely:
+
+```swift
+func makeDinner() async -> Meal {
+  // Create a task group to scope the lifetime of our three child tasks
+  return try await withThrowingTaskGroup(of: CookingTask.self) { group in
+    // spawn three cooking tasks and execute them in parallel:
+    group.spawn {
+      CookingTask.veggies(try await chopVegetables())
+    }
+    group.spawn {
+      CookingTask.meat(await marinateMeat())
+    }
+    group.spawn {
+      CookingTask.oven(await preheatOven(temperature: 350))
+    }
+
+    // prepare variables to collect the results
+    var veggies: [Vegetable]? = nil
+    var meat: Meat? = nil
+    var oven: Oven? = nil
+
+    // collect the results
+    for try await task in group {
+      switch task {
+      case .veggies(let v):
+        veggies = v
+      case .meat(let m):
+        meat = m
+      case .oven(let o):
+        oven = o
+      }
+    }
+
+    // ensure every variable was initialized as expected
+    assert(veggies != nil)
+    assert(meat != nil)
+    assert(oven != nil)
+
+    // prepare the ingredients
+    var ingredients: [Ingredient] = veggies!
+    ingredients.append(meat!)
+
+    // and, finally, cook the meal, awaiting inside the group
+    let dish = Dish(ingredients: ingredients)
+    return try await oven!.cook(dish, duration: .hours(3))
+  }
+}
+```
+
+The `withTaskGroup` scope explicitly delineates any potential concurrency, because it guarantees that any child tasks spawned within it are awaited on as the group scope exits. Any results can be collected using iterating through the group. Errors and cancellation are handled automatically for us by the group.
+
+However, this example showcases the weaknesses of the TaskGroups very well: heterogenous result processing and variable initialization become very boiler plate heavy. While there exist ideas to make this boiler plate go away in future releases, with smarter analysis and type checking, the fundamental issue remains. 
+
+If we step back a little, we can notice that in the example each child task is really producing a *single value* and returning it back to the *parent task*, which then needs to assamble the pieces and proceed with calling some other function. We achieve this by preparing, and assigning into `Optional` variables dedicated for each of the spawned tasks. This is not ideal, since
```

Comment:
`assamble` → `assemble`
```suggestion
If we step back a little, we can notice that in the example each child task is really producing a *single value* and returning it back to the *parent task*, which then needs to assemble the pieces and proceed with calling some other function. We achieve this by preparing, and assigning into `Optional` variables dedicated for each of the spawned tasks. This is not ideal, since
```